types/chainLock.go

```diff
@@ -0,0 +1,123 @@
+package types
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/tendermint/tendermint/crypto"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+type ChainLock struct {
```

Comment:
`CoreChainLock`?

Reply:
Hmmm, it's a chain lock, no need to say it's core.

Reply:
There are 2 chains, Tendermint and Core. Just to represent that's Core feature, not Tendermint. No?